### PR TITLE
docs: Add MDX docs and examples for Indicator components

### PR DESCRIPTION
**Pull request by jamesfan**
## Summary

Migrates MDX docs and code examples for the following Indicator components from the Canvas Site:

- Banner
- Loading Animation
- Skeleton
- Status Indicator

I've removed most of the content from the READMEs for these components -- they now consist only of a link to the Storybook documentation and a link to the corresponding component page on the Design Site (if it exists).

## To-dos

- [x] Migrate Skeleton MDX
- [x] Prepare PR for Design Site with updated Storybook URLs

### CODE REVIEW

File: modules/react/banner/stories/stories.tsx [hunk 1]
```diff
@@ -1,94 +0,0 @@
...
-
-const handleBannerClick = () => {
-  alert(`onClick triggered`);
-};
-
-storiesOf('Components/Indicators/Banner/React/Alert', module)
-  .addParameters({component: Banner})
-  .addDecorator(withReadme(README))
-  .addDecorator(withKnobs)
-  .add('Full', () => (
```
Comment (anicholls) [inline modules/react/banner/stories/stories.tsx:18]: Reminder that all of these story names have changed so we'll have to update the URLs on the design site.

File: modules/react/loading-animation/stories/stories.tsx [hunk 2]
```diff
@@ -1,16 +0,0 @@
...
-import {storiesOf} from '@storybook/react';
-import withReadme from 'storybook-readme/with-readme';
-
-import {LoadingDots} from '../index';
-import README from '../README.md';
-
-storiesOf('Components/Indicators/Loading Animation/React', module)
-  .addParameters({component: LoadingDots})
-  .addDecorator(withReadme(README))
-  .add('Loading Dots', () => (
```
Comment (anicholls) [inline modules/react/loading-animation/stories/stories.tsx:12]: This URL will need to be updated on the design site.

File: modules/react/status-indicator/stories/stories.tsx [hunk 3]
```diff
@@ -1,107 +0,0 @@
...
-import withReadme from 'storybook-readme/with-readme';
-import {uploadCloudIcon} from '@workday/canvas-system-icons-web';
-
-import StatusIndicator from '../index';
-import README from '../README.md';
-
-storiesOf('Components/Indicators/Status Indicator/React', module)
-  .addParameters({component: StatusIndicator})
-  .addDecorator(withReadme(README))
-  .add('Default', () => (
```
Comment (anicholls) [inline modules/react/status-indicator/stories/stories.tsx:13]: These stories will have to be updated on the design site.

**Review by anicholls — APPROVED** [review]
Everything that you've done looks good 👍  I'll await the skeleton docs.

File: modules/react/skeleton/lib/skeleton.tsx [hunk 4]
```diff
@@ -4,11 +4,11 @@ import styled from '@emotion/styled';
 
 import {accessibleHide, createComponent, StyledType} from '@workday/canvas-kit-react/common';
 
-import SkeletonHeader from './parts/skeletonHeader';
-import SkeletonShape from './parts/skeletonShape';
-import SkeletonText from './parts/skeletonText';
+import {SkeletonHeader} from './parts/skeletonHeader';
+import {SkeletonShape} from './parts/skeletonShape';
+import {SkeletonText} from './parts/skeletonText';
```
Comment (jamesfan) [inline modules/react/skeleton/lib/skeleton.tsx:9]: Initially, `ArgsTable` wasn't working for Skeleton subcomponents (e.g., `<ArgsTable of={Skeleton.Text} />` wasn't displaying anything). After consulting with @NicholasBoll, it turns out we needed to convert the default exports in the Skeleton subcomponents to named exports to get their `ArgsTables` to populate properly. These subcomponents aren't exported by the module so this isn't a breaking change.

File: modules/react/skeleton/lib/skeleton.tsx [hunk 5]
```diff
@@ -4,11 +4,11 @@ import styled from '@emotion/styled';
...
 
-interface SkeletonProps {
+export interface SkeletonProps {
```
Comment (jamesfan) [inline modules/react/skeleton/lib/skeleton.tsx:11]: Also, I had to export `SkeletonProps` in order for the `ArgsTable` to show up for `Skeleton`.

File: modules/react/skeleton/index.ts [hunk 6]
```diff
@@ -1 +1 @@
-export * from './lib/skeleton';
+export {Skeleton} from './lib/skeleton';
```
Comment (jamesfan) [inline modules/react/skeleton/index.ts:1]: I limited the export here to `{Skeleton}` to compensate for the newly exported `SkeletonProps` in `skeleton.tsx`.

**Review by jamesfan — COMMENTED** [review]
@anicholls I've added the Skeleton docs. I had to make a few minor changes to the Skeleton component to get `ArgsTable` to work (more on that below). I'll prep the corresponding Design Site PR with the updated Storybook URLs.

**Review by anicholls — APPROVED** [review]

**Review by anicholls — APPROVED** [review]

**Pull request merged**